Sources/FoundationEssentials/AttributedString/AttributedString+UTF16View.swift

```diff
@@ -0,0 +1,139 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if FOUNDATION_FRAMEWORK
+@_spi(Unstable) internal import CollectionsInternal
+#elseif canImport(_RopeModule)
+internal import _RopeModule
+#elseif canImport(_FoundationCollections)
+internal import _FoundationCollections
+#endif
+
+@available(FoundationPreview 6.2, *)
+extension AttributedString {
+    public struct UTF16View: Sendable {
+        internal var _guts: Guts
+        internal var _range: Range<BigString.Index>
+        internal var _identity: Int = 0
+
+        internal init(_ guts: AttributedString.Guts) {
+            self.init(guts, in: guts.stringBounds)
+        }
+
+        internal init(_ guts: Guts, in range: Range<BigString.Index>) {
+            _guts = guts
+            _range = range
+        }
+
+        public init() {
```

Comment:
Is this intentionally `public`? It wasn't included in the proposal. I don't think you can `init` a `String.UTF16View` either.

Reply:
Good call, this was copied from the `CharacterView` and `UnicodeScalarView` code, but I didn't include it in the proposal since there's no need to ever create this yourself especially since it's immutable - I'll remove this from the implementation